### ISSUE TITLE
Request to add global search

### ISSUE THREAD

**Issue by micush** [TechnitiumSoftware/DnsServer#301]
Please add a global search option. It should search through all the zones, scopes, config settings, etc. Clicking on a search result should take you directly to the item.

**Comment by ShreyasZare** [comment]
Thanks for the post. The search feature request is honestly out of scope. This is since there is no database being used in the server which means that for this feature to work, a code which does indexing will be required to be built that will be too much effort for too little reward. Adding a database too is not feasible since not using a database explicitly was a design decision to allow the server to run cross platform with minimal hardware.

**Comment by micush** [comment]
Hi. The stats dashboard is currently run off of something that resembles a database. Doing the same for search should be similar, no?

When you have a zone or a scope filled with thousands of records and no search utility it becomes unusable to find and modify hosts. "CTRL+F" in the browser window will only get you so far. At that point data management becomes unworkable because the on-disk config files are in a format that cannot be read or modified by normal text editing tools.

Please reconsider this request. For large installations it is a must. Thanks for your time. Much appreciated.

**Comment by ShreyasZare** [comment]
> Hi. The stats dashboard is currently run off of something that resembles a database. Doing the same for search should be similar, no?

The dashboard is showing data that is stored in files. Stats are collected per minute and stored in an hourly file. When the day changes the last day's hourly files are merged into a daily file. So its kind of purpose built database which can only be filtered by time.

> When you have a zone or a scope filled with thousands of records and no search utility it becomes unusable to find and modify hosts. "CTRL+F" in the browser window will only get you so far. At that point data management becomes unworkable because the on-disk config files are in a format that cannot be read or modified by normal text editing tools.
> 
> Please reconsider this request. For large installations it is a must. Thanks for your time. Much appreciated.

Let me think on this for a while. May be something like a simple search without much of indexing can be done with decent performance.

**Comment by CRASH-Tech** [comment]
After 11.5 this is the last problem due to which I cannot use the this avesome product.
I have thousands records, and without search it's unusable.

**Comment by ShreyasZare** [comment]
> After 11.5 this is the last problem due to which I cannot use this awesome product. I have thousands records, and without search it's unusable.

I understand your requirement. It would take a while to add such a feature due to time constraints. Also, can you give an estimate number of records you have on the max side so that when developing this feature, I can plan tests that use same number of records to check for performance.

**Comment by CRASH-Tech** [comment]
> > After 11.5 this is the last problem due to which I cannot use this awesome product. I have thousands records, and without search it's unusable.
> 
> I understand your requirement. It would take a while to add such a feature due to time constraints. Also, can you give an estimate number of records you have on the max side so that when developing this feature, I can plan tests that use same number of records to check for performance.

27к records is my current max

**Comment by ShreyasZare** [comment]
> 27к records is my current max

Thanks for the details. Will see if a client side search feature works better for such a zone since all records are already present on the client side.

**Comment by ShreyasZare** [comment]
Technitium DNS Server v13.5 is now available that adds filtering option for zone edit view. Do update and let me know your feedback.

**Comment by micush** [comment]
This works good. Thank you so much. :)

**Comment by ShreyasZare** [comment]
> This works good. Thank you so much. :)

Thanks for the feedback!

**Comment by micush** [comment]
After using it a bit, partial name or wildcard searches would be very useful.

**Comment by ShreyasZare** [comment]
> After using it a bit, partial name or wildcard searches would be very useful.

Ya, will be updating this in coming updates based on feedback.

**Comment by micush** [comment]
The record wildcard search in 13.6 works great. Can you implement the same for the domain search, for those of us with many zones, as well?

**Comment by ShreyasZare** [comment]
> The record wildcard search in 13.6 works great. Can you implement the same for the domain search, for those of us with many zones, as well?

Yes, its planned. It will be added step by step with upcoming updates.

**Comment by Zebble** [comment]
> The record wildcard search in 13.6 works great. Can you implement the same for the domain search, for those of us with many zones, as well?

Would also love to see this.  We have 100+ domains and looking to switch to Technitium.  I may take a crack at a PR if this isn't already in progress?

**Comment by Zebble** [comment]
I (through Claude Code) added a "Search zones..." input box to the Zones tab.  Seems to work well for my purposes. 
(https://github.com/TechnitiumSoftware/DnsServer/pull/1974)(PR#1974)
